core/src/main/scala/org/apache/spark/storage/ShuffleBlockFetcherIterator.scala

```diff
@@ -334,6 +353,7 @@ final class ShuffleBlockFetcherIterator(
                     s"due to Netty OOM, will retry")
                 }
                 remainingBlocks -= blockId
+                logFetchOnCompletionIfSlow()
```

Comment:
If we are calling `logFetchOnCompletionIfSlow` from here, we should recompute what is `totalRequestSize` and `blockCount` - since fetch failed before all blocks were fetched.
Essentially, we need to compute the stats based on infoMap.keys -- remainingBlocks